# This patch applies the following fixes:
#  * allow CXXFLAGS addjustment
#  * generate and install man page
# Copyright (C) 2017-2020 Georgi D. Sotirov <gdsotirov@dir.bg>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
diff -urNad cppcheck-1.90-orig/Makefile cppcheck-1.90/Makefile
--- cppcheck-1.90-orig/Makefile	2019-12-21 12:51:10.000000000 +0200
+++ cppcheck-1.90/Makefile	2020-06-14 19:11:21.827233146 +0300
@@ -100,6 +100,8 @@
 endif
 ifndef CXXFLAGS
     CXXFLAGS=-std=c++0x -O2 -DNDEBUG -Wall -Wno-sign-compare
+else
+    CXXFLAGS+=-std=c++0x -O2 -DNDEBUG -Wall -Wno-sign-compare
 endif
 
 # Increase stack size for Cygwin builds to avoid segmentation fault in limited recursive tests.
@@ -146,10 +148,11 @@
     INCLUDE_FOR_TEST=-Ilib -Icli -isystem externals/simplecpp -isystem externals/tinyxml
 endif
 
-BIN=$(DESTDIR)$(PREFIX)/bin
+BIN=$(PREFIX)/bin
+MAN=$(PREFIX)/man
 
 # For 'make man': sudo apt-get install xsltproc docbook-xsl docbook-xml on Linux
-DB2MAN?=/usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl
+DB2MAN?=/etc/asciidoc/docbook-xsl/manpage.xsl
 XP=xsltproc -''-nonet -''-param man.charmap.use.subset "0"
 MAN_SOURCE=man/cppcheck.1.xml
 
@@ -286,7 +289,7 @@
 cppcheck: $(LIBOBJ) $(CLIOBJ) $(EXTOBJ)
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -o $@ $^ $(LIBS) $(LDFLAGS) $(RDYNAMIC)
 
-all:	cppcheck testrunner
+all:	cppcheck testrunner man
 
 testrunner: $(TESTOBJ) $(LIBOBJ) $(EXTOBJ) cli/threadexecutor.o cli/cmdlineparser.o cli/cppcheckexecutor.o cli/filelister.o
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -o $@ $^ $(LIBS) $(LDFLAGS) $(RDYNAMIC)
@@ -315,16 +318,16 @@
 
 man/cppcheck.1:	$(MAN_SOURCE)
 
-	$(XP) $(DB2MAN) $(MAN_SOURCE)
+	$(XP) --output $@ $(DB2MAN) $(MAN_SOURCE)
 
 tags:
 	ctags -R --exclude=doxyoutput --exclude=test/cfg --exclude=test/synthetic cli externals gui lib test
 
-install: cppcheck
-	install -d ${BIN}
-	install cppcheck ${BIN}
-	install htmlreport/cppcheck-htmlreport ${BIN}
+install: cppcheck man/cppcheck.1
 ifdef FILESDIR
+	install -d ${DESTDIR}${BIN}
+	install cppcheck ${DESTDIR}${BIN}
+	install htmlreport/cppcheck-htmlreport ${DESTDIR}${BIN}
 	install -d ${DESTDIR}${FILESDIR}
 	install -d ${DESTDIR}${FILESDIR}/addons
 	install -m 644 addons/*.py ${DESTDIR}${FILESDIR}/addons
@@ -332,6 +335,8 @@
 	install -m 644 cfg/*.cfg ${DESTDIR}${FILESDIR}/cfg
 	install -d ${DESTDIR}${FILESDIR}/platforms
 	install -m 644 platforms/*.xml ${DESTDIR}${FILESDIR}/platforms
+	install -d ${DESTDIR}${MAN}/man1
+	install man/cppcheck.1 ${DESTDIR}${MAN}/man1/cppcheck.1
 else
 	$(error FILESDIR must be set!)
 endif
